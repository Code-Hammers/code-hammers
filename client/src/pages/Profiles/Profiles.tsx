import React, { useEffect } from "react";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { fetchProfiles } from "../../features/profiles/profilesSlice";

const Profiles = (): JSX.Element => {
  const dispatch = useAppDispatch();
  const profiles = useAppSelector((state) => state.profiles.profiles);

  useEffect(() => {
    dispatch(fetchProfiles());
  }, [dispatch]);

  return (
    <>
      <div className="min-h-screen bg-gray-100 flex flex-col items-center justify-center">
        <h1 className="text-4xl font-extrabold mb-4">PROFILES</h1>
      </div>
      <div>
        {profiles.map((profiles) => (
          <div key={profiles.user}>{profiles.user}</div>
        ))}
      </div>
    </>
  );
};

export default Profiles;
