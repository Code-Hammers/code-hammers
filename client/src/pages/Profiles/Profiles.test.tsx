import React from "react";
import { create } from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import Profiles from "./Profiles";

interface State {
  profiles: {
    profiles: { user: string }[];
    status: 'idle' | 'loading' | 'failed';
    error: string | null;
  };
}

const mockStore = configureStore<State>([]);
const initialState: State = {
  profiles: {
    profiles: [
      { user: "User1" },
      { user: "User2" }
    ],
    status: 'idle',
    error: null
  }
  },
};

describe("MainPage Component", () => {
  it("renders correctly", () => {
    const store = mockStore(initialState);
    const tree = create(
      <Provider store={store}>
        <Profiles />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

