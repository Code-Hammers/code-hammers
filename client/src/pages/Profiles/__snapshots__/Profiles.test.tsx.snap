// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component renders correctly 1`] = `
[
  <div
    className="min-h-screen bg-gray-100 flex flex-col items-center justify-center"
  >
    <h1
      className="text-4xl font-extrabold mb-4"
    >
      PROFILES
    </h1>
  </div>,
  <div>
    <a
      href="/app/profile/User1"
      onClick={[Function]}
    >
      <div
        className="bg-gray-100 flex flex-col items-center justify-center p-4 rounded-lg shadow-lg"
      >
        <h1
          className="text-4xl font-extrabold mb-4"
        />
        <h2
          className="text-4xl font-extrabold mb-4"
        />
      </div>
    </a>
    <a
      href="/app/profile/User2"
      onClick={[Function]}
    >
      <div
        className="bg-gray-100 flex flex-col items-center justify-center p-4 rounded-lg shadow-lg"
      >
        <h1
          className="text-4xl font-extrabold mb-4"
        />
        <h2
          className="text-4xl font-extrabold mb-4"
        />
      </div>
    </a>
  </div>,
]
`;
